bin/cobbler-settings

```diff
@@ -114,15 +116,40 @@ def modify(args: argparse.Namespace) -> int:
     schema = migrations.get_schema(migrations.get_installed_version())
     setting_obj = helper.Setting(args.key, args.value)
     # _schema used due to old version of python3-schema in Leap
-    key_type = schema._schema[setting_obj.key_name]
+    try:
+        key_type = schema._schema[setting_obj.key_name]
+    except KeyError:
+        key_found = False
+        for entry in schema._schema:
+            if isinstance(entry, Optional):
+                if entry._schema == setting_obj.key_name:
+                    key_type = schema._schema[entry]
+                    key_found = True
+                    break
+        if not key_found:
+            print("Requested key not found in the settings!")
+            return 1
+    if key_type not in (bool, int, float, str, dict, list):
+        if isinstance(key_type, list):
+            key_type = list
+        elif isinstance(key_type, dict):
+            key_type = dict
+        else:
+            print("Unsupported type!")
+    if isinstance(Optional, key_type):
+        key_type = schema._schema[setting_obj.key_name].name
     if key_type == bool:
-        setting_obj.value = input_boolean(setting_obj.value)
+        setting_obj.value = utils.input_boolean(setting_obj.value)
     elif key_type == int:
         setting_obj.value = int(setting_obj.value)
+    elif key_type == float:
+        setting_obj.value = float(setting_obj.value)
     elif key_type == list:
-        setting_obj.value = input_string_or_list(setting_obj.value)
+        setting_obj.value = utils.input_string_or_list(setting_obj.value)
     elif key_type == dict:
-        setting_obj.value = input_string_or_dict(setting_obj.value)
+        setting_obj.value = utils.input_string_or_dict(setting_obj.value)
+    elif key_type == str:
+        setting_obj.value = setting_obj.value
     else:
         print("Unsupported type!")
```

Comment:
From reading the function, this print call is done twice, right? Wouldn't it make sense to return the first time?

Reply:
I added a `return` statement and it doesn't seem to affect the overall functionality according to our tests.